Codecademy/gamut
# feat(CourseCard): New course card

### Overview


This PR adds the existing [CourseCard](https://github.com/codecademy-engineering/mono/blob/develop/apps/portal-app/src/components/CatalogPage/CourseCard.tsx) component into gamut-labs. 

[This PR](https://github.com/codecademy-engineering/mono/pull/1136) removes it from mono and replaces it with this gamut-labs version. 

[Related gamut PR for context ](https://github.com/Codecademy/gamut/pull/2653)



### PR Checklist

- [x] Related to designs: https://www.figma.com/file/q0bRIpoO92YfCpy4nCiQGI/What's-your-goal---iteration?node-id=702%3A3308&t=GjAsG8u5UGR3D1jO-0
- [x] Related to JIRA ticket: [DISC-1043]

#### Testing Instructions

1. Open the latest Styleguide deploy preview below and search for CourseCard
2. Review this page and make sure everything is captured for this new course card 
3. Open the monorepo PR env link and see that the course cards on catalog, hubs, and full catalog page look the same as production. 

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/34458) 
| Monorepo       | [Portal PR](https://github.com/codecademy-engineering/mono/pull/1136)  | [Monorepo Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-1136)   




[DISC-1040]: https://codecademy.atlassian.net/browse/DISC-1040?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DISC-1043]: https://codecademy.atlassian.net/browse/DISC-1043?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
